api/billingd/filrewards/filrewards.go

```diff
@@ -0,0 +1,217 @@
+package filrewards
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/textileio/textile/v2/api/billingd/analytics"
+	mdb "github.com/textileio/textile/v2/mongodb"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type Event int
+
+const (
+	Unspecified Event = iota
+	FirstKeyAccountCreated
+	FirstKeyUserCreated // does this fit "register first user?"
+	FirstOrgCreated
+	InitialBillingSetup
+	FirstBucketCreated
+	FirstBucketArchiveCreated
+	FirstMailboxCreated
+	FirstThreadDbCreated
+)
+
+const baseAttoFILReward = 1000 // What should this be? Should we read it from mongo?
+
+// maybe we want to read the meta values from mongo so we can update live.
+var eventRewards = map[Event]reward{
+	FirstKeyAccountCreated:    {factor: 3},
+	FirstKeyUserCreated:       {factor: 1},
+	FirstOrgCreated:           {factor: 3},
+	InitialBillingSetup:       {factor: 1},
+	FirstBucketCreated:        {factor: 2},
+	FirstBucketArchiveCreated: {factor: 2},
+	FirstMailboxCreated:       {factor: 1},
+	FirstThreadDbCreated:      {factor: 1},
+}
+
+type reward struct {
+	factor int
+}
+
+type RewardRecord struct {
+	Key               string          `bson:"key"`
+	AccountType       mdb.AccountType `bson:"account_type"`
+	Event             Event           `bson:"event"`
+	Factor            int             `bson:"factor"`
+	BaseAttoFILReward int             `bson:"base_atto_fil_reward"`
+	CreatedAt         time.Time       `bson:"created_at"`
+}
```

Comment:
This represents a historical record of an achieved reward.

Reply:
I record the `Factor` and `BaseAttoFILReward` at the time the reward was achieved just in case we want to use that information. Maybe we want to use whatever the current values are, but at least we have the option to use the old values that are recorded here.